tests/llm/test_rag.py

```diff
@@ -0,0 +1,119 @@
+import re
+import pytest
+from langchain_core.documents import Document
+from langchain_core.vectorstores import InMemoryVectorStore
+from langchain_community.retrievers import BM25Retriever
+from unittest.mock import AsyncMock, Mock
+
+from chatsky import Pipeline
+from chatsky.llm.rag import get_documents
+from chatsky.utils.testing import TOY_SCRIPT
+
+
+@pytest.fixture
+def pipeline_with_retrievers():
+    high_score_test_doc = Document(id="1", metadata={}, page_content="High score test doc")
+    low_score_test_doc = Document(id="2", metadata={}, page_content="Low score test doc")
+    vec_store = Mock(spec=InMemoryVectorStore)
+    vec_store.asimilarity_search_with_score = AsyncMock(
+        return_value=[(high_score_test_doc, 0.9), (low_score_test_doc, 0.2)]
+    )
+
+    just_retriever = Mock(spec=BM25Retriever)
+    just_retriever.invoke = Mock(return_value=[high_score_test_doc, low_score_test_doc])
```

Comment:
Why pass concrete specifications to `Mock` instead of base classes (i.e. `VectorStore` and `BaseRetriever`)?
If base classes are used we don't have to add `langchain_community` as a dependency.

Another way to do this is to create custom subclasses of `VectorStore` and `BaseRetriever` for the tests.